core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -0,0 +1,157 @@
+                            @Override
+                            public void onNext(Frame frame) {
+                                if (frame != null && new String(frame.getPayload()).contains("Connected")) {
+                                    latch.countDown();
+                                }
+                            }
+                        };
+
+                        try (
+                                Closeable __ = dockerClient.logContainerCmd(containerId)
+                                        .withFollowStream(true)
+                                        .withSince(0)
+                                        .withStdErr(true)
+                                        .exec(callback)
+                        ) {
+                            return latch.await(1, TimeUnit.SECONDS);
+                        }
+                    });
+                } catch (TimeoutException e) {
+                    throw new ContainerLaunchException("Timed out waiting for log output", e);
+                }
+            }
+        });
+    }
+
+    public VncRecordingContainer withVncPassword(@NonNull String vncPassword) {
+        this.vncPassword = vncPassword;
+        return this;
+    }
+
+    public VncRecordingContainer withVncPort(int vncPort) {
+        this.vncPort = vncPort;
+        return this;
+    }
+
+    public VncRecordingContainer withFrameRate(int frameRate) {
+        this.frameRate = frameRate;
+        return this;
+    }
+
+    @Override
+    protected void configure() {
+        withCreateContainerCmdModifier(it -> it.withEntrypoint("/bin/sh"));
+
+        if (getNetwork() == null) {
+            withNetwork(Network.newNetwork());
+        }
+
+        String alias = "vnchost-" + Base58.randomString(8);
+        dockerClient.connectToNetworkCmd()
+                .withContainerId(targetContainerIdSupplier.get())
+                .withNetworkId(getNetwork().getId())
+                .withContainerNetwork(new ContainerNetwork().withAliases(alias))
+                .exec();
+
+        setCommand(
+                "-c",
+                "echo '" + Base64.encodeBase64String(vncPassword.getBytes()) + "' | base64 -d > /vnc_password && " +
+                "flvrec.py -o " + RECORDING_FILE_NAME + " -d -r " + frameRate + " -P /vnc_password " + alias + " " + vncPort
+        );
+    }
+
+    @Override
+    public void stop() {
+        try {
+            dockerClient.disconnectFromNetworkCmd()
+                    .withContainerId(targetContainerIdSupplier.get())
+                    .withNetworkId(getNetwork().getId())
+                    .withForce(true)
+                    .exec();
+        } catch (Exception e) {
+            logger().warn("Failed to disconnect container with id '{}' from network '{}'", targetContainerIdSupplier.get(), getNetwork().getId(), e);
+        }
+
+        super.stop();
+    }
+
+    @SneakyThrows
+    public InputStream streamRecord() {
```

Comment:
Just a wording thing, but please could this be `streamRecording`? The same for `saveRecordToFile` (->`saveRecordingToFile`)